src/mono/mono/metadata/handle.h

```diff
@@ -367,6 +367,9 @@ typedef struct _MonoTypeofCastHelper *MonoTypeofCastHelper; // a pointer type un
 
 #define MONO_HANDLE_CAST(type, value) (MONO_HANDLE_CAST_FOR (type) ((value).__raw))
 
+/* Simpler version if the handle is not used */
+#define MONO_HANDLE_PIN(object) MONO_HANDLE_NEW (MonoObject, (object))
```

Comment:
If this only works for `MonoObject`s, the macro name should probably have that at the end. I'm assuming that's the case since you didn't use it above for the `MonoArray`?

Reply:
It should work for all objects, we might need to put casts into it in the future to make that happen.

Reply:
Think I'd rather just put the casts in and use for the `MonoArray` in this PR then, but it's not a huge deal.